exasol/toolbox/nox/_documentation.py

```diff
@@ -88,6 +100,82 @@ def clean_docs(_session: Session) -> None:
         shutil.rmtree(docs_folder)
 
 
+@nox.session(name="docs:links", python=False)
+def docs_list_links(session: Session) -> None:
+    """List all the links within the documentation."""
+    with tempfile.TemporaryDirectory() as path:
+        tmpdir = Path(path)
+        sp = subprocess.run(
+            [
+                "sphinx-build",
+                "-b",
+                "linkcheck",
+                "-D",
+                "linkcheck_ignore=.*",
+                PROJECT_CONFIG.root / "doc",
+                tmpdir,
+            ],
+        )
+        print(sp.returncode)
+        if sp.returncode >= 2:
+            print(sp.stderr)
+            session.error(2)
+        output = tmpdir / "output.json"
+        links = output.read_text().split("\n")
+        file_links = []
+        for link in links:
+            if link != "":
+                line = json.loads(link)
+                if not line["uri"].startswith("#"):
+                    file_links.append(line)
+        file_links.sort(key=lambda file: file["filename"])
+        print(
+            "\n".join(
+                f"filename: {fl['filename']} -> uri: {fl['uri']}" for fl in file_links
+            )
+        )
+
+
+@nox.session(name="docs:links:check", python=False)
+def docs_links_check(session: Session) -> None:
+    """Checks whether all links in the documentation are accessible."""
+    parser = argparse.ArgumentParser(
+        prog="nox -s release:prepare",
+        usage="nox -s release:prepare -- [-h] [-o |--output]",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "-o", "--output", type=Path, help="path to output file", default=None
+    )
+    args = parser.parse_args(session.posargs)
+    with tempfile.TemporaryDirectory() as path:
+        tmpdir = Path(path)
+        sp = subprocess.run(
+            [
+                "sphinx-build",
+                "-b",
+                "linkcheck",
+                PROJECT_CONFIG.root / "doc",
+                tmpdir,
+            ],
+        )
+        print(sp.returncode)
+        if sp.returncode >= 2:
+            print(sp.stderr)
+            session.error(2)
+        if args.output:
+            result_json = tmpdir / "output.json"
+            dst = Path(args.output) / "link-check-output.json"
+            shutil.copyfile(result_json, dst)
+            print(f"file generated at path: {result_json.resolve()}")
+        result_txt = tmpdir / "output.txt"
+        if sp.returncode == 1 or result_txt.read_text() != "":
```

Comment:
maybe something like this -> separate errors visually with a character like `- `and use a walrus operator in some way so that not doing read_text() twice. If you use a walrus operator then, it should be the first item in the or condition, as otherwise, it may not be set later.
```suggestion
        if errors:=result_txt.read_text().splitlines() or sp.returncode == 1:
            escape_red = "\033[31m"
            print(escape_rot + "errors:")
            print('\n'.join(f"- {item}" for item in errors)) 
            session.error(1)
```